datahub-graphql-core/src/main/resources/entity.graphql

```diff
@@ -1719,11 +1719,15 @@ type Role implements Entity {
 
 }
 
+input UserFilter {
+    userUrn: String!
+}
+
 type Actor {
     """
     List of users for which the role is provisioned
     """
-    users: [RoleUser!]
+    users(input: UserFilter): [RoleUser!]
```

Comment:
what if, instead of putting a filter here, we create a new shortcut field for getting whether or not a given role is simply assigned to "me" (the querying user)

what that could look like is a new field on the `Role` type right above called something like `isAssignedToMe: Boolean!`

this field can then have its own custom resolver similar to what you've done above. something very similar (even with the same name I suggest above) is the `IsFormAssignedToMeResolver` for our documentation forms feature. [in GMSGraphQLEngine we have this resolver defined](https://github.com/acryldata/datahub-fork/blob/acryl-main/datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java#L2625) under the `FormActorAssignment` parent. We could do something similar but with the `Role` parent, and have a new resolver that can get the role and the requesting user and check to see if that specific user is in the role.

the logic would be very similar to what we do in `IsFormAssignedToMeResolver` but we could ignore the groups piece.

Therefore it would be much more performant since we could simply check the list from the `Role` object and see if the logged in user is in there and return a boolean yes/no. Then if we stop querying for the list of users from the frontend, we should be in good shape since we would then not hydrate all of these users.

Reply:
or the field could be `userHasAccess` or something

Reply:
Hi Chris, thanks for the feedback. I am aware that the solution would not return the appropriate users, just wanted to get some feedback on what would be the best way to approach the problem seeing that I'm not super familiar with the code base. The `isAssignedToMe` suggestion sounds good (I was trying to achieve something similar, i.e.: not hydrating all the users of the role if not necessary) 👍 I'll try to get that implemented.

Reply:
okay sounds great thanks so much! yeah sorry I didn't realize it was more of a placeholder to start the conversation / get feedback. but it served its job well! let me know if you need any assistance or anything :)